File: android/app/src/main/java/com/echo/dictation/ai/PromptTemplate.kt
package com.echo.dictation.ai

/**
 * All prompt template categories.
 *
 * Adding a new category requires only adding an enum value here — no
 * business-logic changes are needed.
 */
enum class PromptCategory {
    GENERAL,
    BUSINESS,
    PRODUCTIVITY,
    COMMUNICATION,
}

/**
 * A reusable, immutable prompt template that can be applied to any transcript.
 *
 * @param id            Stable identifier used as a key in the prompt library.
 * @param title         Short user-facing label (e.g. "Meeting Notes").
 * @param description   One-line description shown in the UI picker.
 * @param systemPrompt  Instruction sent as the LLM system message.
 * @param category      Grouping for the library UI.
 */
data class PromptTemplate(
    val id: String,
    val title: String,
    val description: String,
    val systemPrompt: String,
    val category: PromptCategory,
)

File: EchoCore/Sources/EchoCore/Services/AI/PromptTemplate.swift
//
//  PromptTemplate.swift
//  EchoCore
//
//  Mirrors Android's domain.ai.PromptTemplate data class and PromptCategory enum.
//

import Foundation

// MARK: - PromptCategory

public enum PromptCategory: String, Sendable, CaseIterable {
    case general        = "GENERAL"
    case business       = "BUSINESS"
    case productivity   = "PRODUCTIVITY"
    case communication  = "COMMUNICATION"
}

// MARK: - PromptTemplate

public struct PromptTemplate: Sendable {
    public let id: String
    public let title: String
    public let description: String
    public let category: PromptCategory
    public let systemPrompt: String
    /// The VersionType this template produces.
    public let targetVersionType: VersionType

    public init(
        id: String,
        title: String,
        description: String,
        category: PromptCategory,
        systemPrompt: String,
        targetVersionType: VersionType
    ) {
        self.id = id
        self.title = title
        self.description = description
        self.category = category
        self.systemPrompt = systemPrompt
        self.targetVersionType = targetVersionType
    }
}
